### PR TITLE
[EASI-3671] Update precedence rules (backend)

### PR DESCRIPTION
**Pull request by OddTomBrooks**
# EASI-3671

## Changes and Description

- Updated gql schema
- Created flyway migration adding new enum type and table columns
- Updated data model
- Updated Postman collection

## How to test this change

- Bring servers up
- Verify migration succeeded
- Run plan beneficiaries update endpoint in Postman
- Verify success and data in response

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [x] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by StevenWadeOddball — CHANGES_REQUESTED** [review]
@OddTomBrooks, please update the server tests, and request review when the tests are passing.

File: migrations/V118__Update_Precedence_Rules.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TYPE YES_NO_FILTER AS ENUM (
```
Comment (StevenWadeOddball) [inline migrations/V118__Update_Precedence_Rules.sql:1]: Non blocking, but filter feels a bit misleading here. It's not really a filter, it's a question answer. Maybe `YES_NO_TYPE` would be more consistent?
```suggestion
CREATE TYPE YES_NO_TYPE AS ENUM (
```

**Review by StevenWadeOddball — CHANGES_REQUESTED** [review]
Thanks @OddTomBrooks , the structure of this look good. The migration of legacy data needs to be handled still.

I also left a note for naming, and for the postman collection not functioning as expected (field names are off)

File: MINT.postman_collection.json [hunk 2]
```diff
@@ -298,7 +298,7 @@
...
...leDifferentNote\n        excludeCertainCharacteristics\n        excludeCertainCharacteristicsCriteria\n        excludeCertainCharacteristicsNote\n        numberPeopleImpacted\n        estimateConfidence\n        confidenceNote\n        beneficiarySelectionMethod\n        beneficiarySelectionOther\n        beneficiarySelectionNote\n        beneficiarySelectionFrequency\n        beneficiarySelectionFrequencyOther\n        beneficiarySelectionFrequencyNote\n        beneficiaryOverlap\n        beneficiaryOverlapNote\n        precedenceRulesExist\n        precedenceRulesExistYes\n        precedenceRulesExistNo\n        precedenceRules\n        createdBy\n        createdDts\n        modifiedBy\n        modifiedDts\n        readyForReviewBy\n        readyForReviewDts\n        readyForClearanceBy\n        readyForClearanceDts\n        status\n\n        createdByUserAccount \n        {\n            commonName\n        }   \n        modifiedByUserAccount\n        {\n            commonName\n        }     \n        readyForReviewByUserAccount \n        {\n            commonName\n        }\n        readyForClearanceByUserAccount \n        {\n            commonName\n        }        \n    }\n}",
```
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply MINT.postman_collection.json:301]: I'll take a look. I thought I updated postman last week but probably forgot to export it before committing.

File: MINT.postman_collection.json [hunk 3]
```diff
@@ -277,8 +277,8 @@
...
...gainsharePayments\n        gainsharePaymentsTrack\n        gainsharePaymentsEligibility\n        gainsharePaymentsEligibilityOther\n        gainsharePaymentsNote\n        participantsIds\n        participantsIdsOther\n        participantsIDSNote\n        providerAdditionFrequency\n        providerAdditionFrequencyOther\n        providerAdditionFrequencyNote\n        providerAddMethod\n        providerAddMethodOther\n        providerAddMethodNote\n        providerLeaveMethod\n        providerLeaveMethodOther\n        providerLeaveMethodNote\n        providerOverlap\n        providerOverlapHierarchy\n        providerOverlapNote\n        createdBy\n        createdDts\n        modifiedBy\n        modifiedDts\n        readyForReviewBy\n        readyForReviewDts\n        readyForClearanceBy\n        readyForClearanceDts\n        \n        createdByUserAccount \n        {\n            commonName\n        }   \n        modifiedByUserAccount\n        {\n            commonName\n        }     \n        readyForReviewByUserAccount \n        {\n            commonName\n        }\n        readyForClearanceByUserAccount \n        {\n            commonName\n        }\n\n\n        status\n    }\n}",
```
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply MINT.postman_collection.json:280]: Ditto above, I tested things in Postman to validate the work so likely just need to export the collection

File: migrations/V118__Update_Precedence_Rules.sql (hunk 1 above)
Comment (OddTomBrooks) [reply migrations/V118__Update_Precedence_Rules.sql:1]: That makes sense. I was unsure of the naming for this one, I think your suggestion is a good one.

File: migrations/V118__Update_Precedence_Rules.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  'YES',
+  'NO'
+  );
+
+ALTER TABLE plan_beneficiaries
```
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply migrations/V118__Update_Precedence_Rules.sql:6]: Ok I'll take a stab at this

**Pull request merged**